File: Easy_python/415_add_strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        returnlist=['0']*(max(len(num1),len(num2))+1)
        if len(num1)<len(num2):
            num1=num1.zfill(len(num2)+1)
            num2=num2.zfill(len(num2)+1)
        else:
            num2=num2.zfill(len(num1)+1)
            num1=num1.zfill(len(num1)+1)
        last=0
        for i in reversed(range(len(num1))):
            
            if int(num1[i])+int(num2[i])+last>=10:
                returnlist[i]=str(int(num1[i])+int(num2[i])+last-10)
                last=1
            else:
                returnlist[i]=str(int(num1[i])+int(num2[i])+last)
                last=0
        #print(returnlist)
        if returnlist[0]=="0":
            return ''.join(returnlist[1:])
        else:
            return ''.join(returnlist)
               

File: Easy/415_Add_strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        //到序相加，进位数字保存
        string res = "";
        int i = num1.length()-1 ,j = num2.length()-1 , tempsum = 0, add = 0;
        while(i >= 0 || j >= 0 || add !=0)
        {
            
            tempsum = (i>= 0 ? (num1[i]-'0') % 10:0) + (j>= 0 ?(num2[j]-'0') % 10 :0) + add;
            if(tempsum / 10 == 1)
            {
                add = 1;
            }
            else
            {
                 add = 0;
            }
            res += tempsum % 10 +'0';
            
            i--;
            j--;
        }
        reverse(res.begin(),res.end());
        return res;
    }
};
